pkg/controllers/monitoring-stack/controller.go

```diff
@@ -122,6 +143,16 @@ func (r *reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 		return ctrl.Result{}, err
 	}
 
+	requeue, err := r.createGrafanaDSWatch(ctx)
+
+	if requeue {
+		return ctrl.Result{RequeueAfter: 5 * time.Second}, nil
+	}
+
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
```

Comment:
*nit:* always `err` check first 
Go also allows of this to be written succinctly while keeping the scope of these variables minimal as follows 

```go 

if requeue, err := r.createGrafanaDSWatch(ctx); err != nil {
  return ctrl.Result{}, err
} else if requeue {
   return ctrl.Result{RequeueAfter: 5 * time.Second}, nil
}
```